File: Python/206.py
"""
@no 206
@name Reverse Linked List
"""
class Solution:
    def reverseList(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        if not head: return None
        elif not head.next: return head
        else:
            p = head
            q = head.next
            while q.next:
                t = q.next
                q.next = p
                p = q
                q = t
            q.next = p
            head.next = None
            return q

File: Java/206.java
/**
 * @no 206
 * @name Reverse Linked List
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        ListNode t = reverseList(head.next);
        head.next.next = head;
        head.next = null;
        return t;
    }
}